app-ts-2/src/App.tsx

```diff
@@ -2,20 +2,89 @@ import * as React from 'react';
 import { Component } from 'react';
 import './App.css';
 import ProductList from './ProductList';
+import { Product } from './Models/Product';
 import { GetData } from './data';
 
-const products = GetData();
+interface AppState {
+  products: Product[]; // imported from `/Models/Product.ts
+  newProductName: string;
+  newProductDescription: string;
+}
+
+interface AppProps {
+}
+
+class App extends Component<{}, AppState> {
+
+  constructor(props: AppProps, state: AppState) {
+    super(props);
+    this.state = {
+      products: GetData(),
+      newProductName: '',
+      newProductDescription: ''
+    };
+    this.onNameChange = this.onNameChange.bind(this);
+    this.onDescriptionChange = this.onDescriptionChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.onRemoveProduct = this.onRemoveProduct.bind(this);
+  }
+
+  onNameChange(event: React.FormEvent<HTMLInputElement>): void {
+    this.setState({
+      newProductName: event.currentTarget.value
```

Comment:
Why `[name]: value` doesn't work?
Is it because it's TypeScript?

It should work according to https://reactjs.org/docs/forms.html#handling-multiple-inputs:
```js
  handleInputChange(event) {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  }
```